CMakeLists.txt

```diff
@@ -371,6 +322,146 @@ RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}"
+	set(MARIADB_CPP_CONNECTOR_DIR "${MARIADB_CONNECTOR_DIR}/MariaDB/MariaDB C++ Connector 64-bit")
+
+	file(MAKE_DIRECTORY "${MARIADB_MSI_DIR}")
+	file(MAKE_DIRECTORY "${MARIADB_CONNECTOR_DIR}")
+
+	if(NOT EXISTS "${MARIADB_MSI_DIR}/mariadb-connector-c-3.2.5-win64.msi" )
+		message("Downloading mariadb connector/c")
+		file(DOWNLOAD https://dlm.mariadb.com/1936366/connectors/c/connector-c-3.2.5/mariadb-connector-c-3.2.5-win64.msi
+			"${MARIADB_MSI_DIR}/mariadb-connector-c-3.2.5-win64.msi"
+			EXPECTED_HASH MD5=09d418c290109068a5bea136dafca36b)
+	endif()
+
+	if(NOT EXISTS "${MARIADB_MSI_DIR}/mariadb-connector-cpp-1.0.1-win64.msi" )
+		message("Downloading mariadb connector/c++")
+		file(DOWNLOAD https://dlm.mariadb.com/1683453/connectors/cpp/connector-cpp-1.0.1/mariadb-connector-cpp-1.0.1-win64.msi
+			"${MARIADB_MSI_DIR}/mariadb-connector-cpp-1.0.1-win64.msi"
+			EXPECTED_HASH MD5=548e743fbf067d21d42b81d958bf4ed7)
+	endif()
+
+
+	file(TO_NATIVE_PATH "${MARIADB_CONNECTOR_DIR}" MSIEXEC_TARGETDIR)
+	# extract msi files without installing to users system
+	if(NOT EXISTS "${MARIADB_C_CONNECTOR_DIR}")
+		file(TO_NATIVE_PATH "${MARIADB_MSI_DIR}/mariadb-connector-c-3.2.5-win64.msi" MSI_DIR)
+		execute_process(COMMAND msiexec /a ${MSI_DIR} /qn TARGETDIR=${MSIEXEC_TARGETDIR})
+	endif()
+	
+	if(NOT EXISTS "${MARIADB_CPP_CONNECTOR_DIR}")
+	file(TO_NATIVE_PATH "${MARIADB_MSI_DIR}/mariadb-connector-cpp-1.0.1-win64.msi" MSI_DIR)
+		execute_process(COMMAND msiexec /a ${MSI_DIR} /qn TARGETDIR=${MSIEXEC_TARGETDIR})
+	endif()
+	
+	set(MARIADB_SHARED_LIBRARY_LOCATION "${MARIADB_CPP_CONNECTOR_DIR}/mariadbcpp.dll")
+	set(MARIADB_IMPLIB_LOCATION "${MARIADB_CPP_CONNECTOR_DIR}/mariadbcpp.lib")
+	set(MARIADB_INCLUDE_DIR "${MARIADB_CPP_CONNECTOR_DIR}/include/mariadb")
+
+	add_custom_target(mariadb_connector_cpp)
+	add_custom_command(TARGET mariadb_connector_cpp POST_BUILD
+					COMMAND ${CMAKE_COMMAND} -E make_directory "${PROJECT_BINARY_DIR}/$<CONFIG>"
+					COMMAND ${CMAKE_COMMAND} -E copy_if_different 
+					"${MARIADB_CPP_CONNECTOR_DIR}/mariadbcpp.dll"
+					"${MARIADB_C_CONNECTOR_DIR}/lib/libmariadb.dll"
+					"${PROJECT_BINARY_DIR}/$<CONFIG>")
+
+	# MariaDB uses plugins that the database needs to load, the prebuilt binaries by default will try to find the libraries in system directories,
+	# so set this define and the servers will set the MARIADB_PLUGIN_DIR environment variable to the appropriate directory.
+	# Plugin directory is determined at dll load time (this will happen before main()) so we need to delay the dll load so that we can set the environment variable
+	add_link_options(/DELAYLOAD:mariadbcpp.dll)
+	add_compile_definitions(MARIADB_PLUGIN_DIR_OVERRIDE="${MARIADB_CPP_CONNECTOR_DIR}/plugin")
+else() # Build from source
+	if(EXISTS "${CMAKE_SOURCE_DIR}/thirdparty/mariadb-connector-cpp/test/")
+		# renaming test folder stops it from being built, saves some time when compiling
```

Comment:
Is this necessary? it dirties the tree and may cause potential issues with updates

Reply:
I notice no noticeable time difference in compile times with or without this, so imo, not worth it with how ti dirties the tree

Reply:
I'm happy to remove it. I think it just took a really long time when I initially set up as I was compiling on a pi

Reply:
I agree with aronwk on the above.  This should be removed imo

Reply:
Why exactly are we switching out this connector anyways?  This seems to just increase compile time and add more functionality than we actively use for the server.  Is this a change that has to be done to support aarch64?

Reply:
I've made a ticket on their jira for this https://jira.mariadb.org/browse/CONCPP-102
In the mean time I don't think the extra compile time is an issue, plus with your speedups, I don' think it matters

Reply:
Add `ignore = dirty` in the submodule definition
https://stackoverflow.com/questions/3240881/git-can-i-suppress-listing-of-modified-content-dirty-submodule-entries-in-sta

And potentially only delete `${CMAKE_SOURCE_DIR}/thirdparty/mariadb-connector-cpp/test/CMakeLists.txt` Since that one file's existence will make tests compile
https://github.com/mariadb-corporation/mariadb-connector-cpp/blob/master/CMakeLists.txt#L508-L514